CRM/Contribute/Form/Task/Invoice.php

```diff
@@ -245,26 +245,17 @@ public static function printPDF($contribIDs, &$params, $contactIds) {
 
       $objects['contribution']->receive_date = CRM_Utils_Date::isoToMysql($objects['contribution']->receive_date);
 
-      $addressParams = ['contact_id' => $contribution->contact_id];
-      $addressDetails = CRM_Core_BAO_Address::getValues($addressParams);
-
-      // to get billing address if present
+      // Fetch the billing address. getValues should prioritize the billing
+      // address, otherwise will return the primary address.
       $billingAddress = [];
-      foreach ($addressDetails as $address) {
-        if (($address['is_billing'] == 1) && ($address['is_primary'] == 1) && ($address['contact_id'] == $contribution->contact_id)) {
-          $billingAddress[$address['contact_id']] = $address;
-          break;
-        }
-        elseif (($address['is_billing'] == 0 && $address['is_primary'] == 1) || ($address['is_billing'] == 1) && ($address['contact_id'] == $contribution->contact_id)) {
-          $billingAddress[$address['contact_id']] = $address;
-        }
-      }
 
-      if (!empty($billingAddress[$contribution->contact_id]['state_province_id'])) {
-        $stateProvinceAbbreviation = CRM_Core_PseudoConstant::stateProvinceAbbreviation($billingAddress[$contribution->contact_id]['state_province_id']);
-      }
-      else {
-        $stateProvinceAbbreviation = '';
+      $addressDetails = CRM_Core_BAO_Address::getValues([
+        'contact_id' => $contribution->contact_id,
+        'is_billing' => 1,
+      ]);
+
```

Comment:
We seems to lose the case where there is no billing but there is a primary address.
It seems useful to keep that possibility for offline invoice creation.